docs/custom-flows/email-password.mdx

```diff
@@ -34,105 +34,104 @@ A successful sign-up consists of the following steps:
-  const [emailAddress, setEmailAddress] = useState("");
-  const [password, setPassword] = useState("");
-  const [pendingVerification, setPendingVerification] = useState(false);
-  const [code, setCode] = useState("");
-  const router = useRouter();
-  // start the sign up process.
-  const handleSubmit = async (e) => {
-    e.preventDefault();
-    if (!isLoaded) {
-      return;
-    }
+<CodeBlockTabs options={['App Router', 'Pages Router']}>
+```tsx filename="app/sign-up/[[...sing-up]]/page.tsx
+'use client'
+
+import * as React from 'react'
+import { useSignUp } from '@clerk/nextjs'
+import { useRouter } from 'next/navigation'
+import { ClerkAPIErrorJSON } from '@clerk/types'
+
+export default function Page() {
+  const { isLoaded, signUp, setActive } = useSignUp()
+  const [emailAddress, setEmailAddress] = React.useState("")
+  const [password, setPassword] = React.useState("")
+  const [verifying, setVerifying] = React.useState(false)
+  const [loading, setLoading] = React.useState(false)
+  const [code, setCode] = React.useState("")
+  const [errors, setErrors] = React.useState<ClerkAPIErrorJSON[] | null>(null)
+  const router = useRouter()
+
+  const handleSubmit = async (e: React.FormEvent) => {
+    e.preventDefault()
+    setLoading(true)
+    if (!isLoaded) return
 
     try {
       await signUp.create({
-        emailAddress,
-        password,
-      });
+        emailAddress, password
+      })
 
-      // send the email.
-      await signUp.prepareEmailAddressVerification({ strategy: "email_code" });
+      await signUp.prepareEmailAddressVerification({
+        strategy: 'email_code'
+      })
 
-      // change the UI to our pending section.
-      setPendingVerification(true);
-    } catch (err: any) {
-      console.error(JSON.stringify(err, null, 2));
+      setVerifying(true)
+      setLoading(false)
     }
-  };
-
-  // This verifies the user using email code that is delivered.
-  const onPressVerify = async (e) => {
-    e.preventDefault();
-    if (!isLoaded) {
-      return;
+    catch (err: any) {
+      console.error('Error:', JSON.stringify(err, null, 2))
+      setErrors(err.errors)
+      setLoading(false)
     }
+  }
+
+  const handleVerify = async (e: React.FormEvent) => {
+    e.preventDefault()
+    setLoading(true)
+    if (!isLoaded) return
 
     try {
       const completeSignUp = await signUp.attemptEmailAddressVerification({
-        code,
-      });
-      if (completeSignUp.status !== "complete") {
-        /*  investigate the response, to see if there was an error
-         or if the user needs to complete more steps.*/
-        console.log(JSON.stringify(completeSignUp, null, 2));
-      }
+        code
+      })
+
       if (completeSignUp.status === "complete") {
         await setActive({ session: completeSignUp.createdSessionId })
-        router.push("/");
       }
-    } catch (err: any) {
-      console.error(JSON.stringify(err, null, 2));
+      setLoading(false)
+      router.push("/")
```

Comment:
Should this be an "else" case, or does this belong in the "if" case, "if the form is completed" ?